compiler/luci-interpreter/include/luci_interpreter/Interpreter.h

```diff
@@ -59,8 +59,14 @@ class Interpreter
   void createKernels(const loco::Graph *graph);
 
   const loco::Graph *_main_graph = nullptr;
-  std::unique_ptr<class TensorMap> _tensor_map;
-  std::unique_ptr<class KernelMap> _kernel_map;
+  std::unique_ptr<class TensorMap> _node_to_tensor;
+
+  // Kernels, in execution order.
+  std::vector<std::unique_ptr<class Kernel>> _kernels;
+
+  // Observer functionality support.
+  std::unique_ptr<struct RuntimeToIR> _runtime_to_ir;
+  std::unique_ptr<class EventNotifier> _event_notifier;
```

Comment:
I didn't get why you need `RuntimeToIR` structure and `EventNotifier`, could you clarify?

Reply:
It is in the description. To make runtime entities independent of IR.
Please see the draft for details.

Reply:
My question was not about purpose of these members, but about purpose of separate structures.
From my point EventNotifier can be inlined, This applies to RuntimeToIR too.

Well, Let's return to this discussion, when everything is working =)